libyul/ObjectOptimizer.cpp

```diff
@@ -158,6 +158,7 @@ std::optional<h256> ObjectOptimizer::calculateCacheKey(
 	rawKey += h256(u256(_settings.expectedExecutionsPerDeployment)).asBytes();
 	rawKey += FixedHash<1>(uint8_t(_isCreation ? 0 : 1)).asBytes();
 	rawKey += keccak256(_settings.evmVersion.name()).asBytes();
+	rawKey += FixedHash<1>(uint8_t(_settings.eofVersion ? 0 : *_settings.eofVersion)).asBytes();
```

Comment:
Please also assert that `_settings.eofVersion` is never `0`, or this would be ambiguous.

Same in any other place that substitutes `0` for `nullopt`. E.g. after my suggestion in `ASTJsonExporter.cpp`.